reach

```diff
@@ -115,10 +115,103 @@ run_s () {
   sh -ac "$TMP/out.sh" "$0"
 }
 
+package_manager () {
+  if ( which apt > /dev/null 2>&1 ); then
+    export PACKAGE_MANAGER="apt"
+  fi
+
+  if ( which yum > /dev/null 2>&1 ); then
+    export PACKAGE_MANAGER="yum"
+  fi
+}
+
+install_make () {
+  package_manager
+
+  sudo ${PACKAGE_MANAGER} install make
+}
+
+install_curl () {
+  package_manager
+
+  sudo ${PACKAGE_MANAGER} install curl 
+}
+
+timeout_validate () {
+  TIMEOUT=$1
+  STOPPER=$2
+  MESSAGE=$3
+
+  if [ "$STOPPER" -eq "$TIMEOUT" ]; then
+    printf "%s" "$MESSAGE"
+    exit 1;
+  fi
+}
+
+install_docker () {
+  PWD=$(pwd)
```

Comment:
Many (most or all?) shells already define `PWD=$(pwd)` so I don't think you should shadow this name.

You could call it something else or just use `cd -` at the end to achieve the same behavior.

Try `env | grep PWD` and this from a fresh terminal:
``` sh
$ cd -
bash: cd: OLDPWD not set

$ cd .. && pwd
/usr/home

$ cd -
/usr/home/matt
```

Reply:
I have renamed all these variables. I tried using OLDPWD, but that one is already used by the shell.
I have renamed it to OPWD.